src/jit/importer.cpp

```diff
@@ -13940,15 +13939,22 @@ bool Compiler::impReturnInstruction(BasicBlock *block, int prefixFlags, OPCODE &
 
         op2 = impAssignStructPtr(retBuffAddr, op2, retClsHnd, (unsigned)CHECK_SPILL_ALL);
         impAppendTree(op2, (unsigned)CHECK_SPILL_NONE, impCurStmtOffs);
-        if (compIsProfilerHookNeeded())
+
+#if !defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+        // As per the System V ABI, the address of RetBuf needs to be returned by
+        // methods with hidden RetBufArg in RAX. In such case GT_RETURN is of TYP_BYREF,
+        // returning the address of RetBuf.
+        if (!compIsProfilerHookNeeded())
```

Comment:
Please restructure this code and/or edit the comments. You are talking about System V ABI in an ifdef that doesn't apply in that case, and then the comment is basically repeated below. Instead, I think you should put a comment above the #if that describes all the cases.